ESPixelStick/src/input/InputFPPRemotePlayList.hpp

```diff
@@ -59,6 +59,24 @@ class c_InputFPPRemotePlayList : public c_InputFPPRemotePlayItem
     c_InputFPPRemotePlayItem * pInputFPPRemotePlayItem = nullptr;
 
     uint32_t PlayListEntryId     = 0;
+
+    // BUGBUG -- time_t creates issues for portable code, and for overflow-safe code
+    // time_t is only required to be a "real" type, which means it can be either a float or an integer.
+    // even when time_t is an integer type, it can be signed or unsigned
+    //
+    // Current code appears to assume that time_t is not a float.
+    //
+    // C++11 does not expose any constant for minimum / maximum value for the time_t type.
+    // This makes it impossible (without type traits) to ensure math operations on time_t types do
+    // not result in undefined behavior (signed types only) and/or overflow (all types).
+    //
+    // Therefore, it is highly recommended to move AWAY from use of time_t wherever possible,
+    // and instead use types that expose their minimum and maximum values, respectively.
+    //
+    // For now, because the code presumes time_t is an integer type, the following assertion
+    // ensures this for all practical purposes.
+    static_assert( (((time_t)1) / 2) == 0 ); // Verify time_t is an integer type (alternative: float)
+
```

Comment:
This code is not intended to be portable. It is tailored to the ESP product line and time_t has well defined semantics and well understood side effects.

Reply:
I'm not as sure as you that `time_t` is so well defined, even for the ESP product line.  Thus, I challenge you to a simple experiment (I will also do this):

Without validating your understanding (e.g., don't check via compilation), write down your answers to the following questions:

1. What is the result of `sizeof(time_t)`?
2. What is the minimum and maximum values that can be stored in `time_t`?
3. Is `time_t` signed or unsigned?
4. What happens when there is overflow on `time_t`?
5. Based on the above answers, would you say that the use of `time_t` currently avoids undefined behavior (UB)?

After you've written answers to the above (don't have to send them), feel free to verify them programmatically, such as by adding static asserts to the code, and compiling the default boards.

Were you surprised by any of the results?

Reply:
for questions 1-3, I would hope `time_t` is a 64-bit **_signed_** value ... equivalent to `int64_t` on both ESP8266 and ESP32 (min/max of `INT64_MIN`/`INT64_MAX`).

for question 4, I would say this is undefined behavior, because I believe `time_t` is (should be) a signed 64-bit value.

Reply:
Update: My answers above are **_wrong_**, and in a significant way.  When I tried to validate my assumptions with the compilers, it showed surprising results.  Those surprise results (imho) support avoiding the use of this type when possible.

I will open a separate discussion on the use of `time_t`....